masakari-controller/controller/masakari_util.py

```diff
@@ -696,6 +698,31 @@ def disable_host_status(self, hostname):
             self.rc_util.syslogout(error_code + msg, syslog.LOG_ERR)
             raise
 
+    def force_host_down_pf9(self, hostname):
+        """
+        Force host service status as "down"
+        :param hostname:
+        :return:
+        """
+        try:
+            self.rc_util.syslogout('Force nova-compute down on %s' % hostname,
+                                   syslog.LOG_INFO)
+            token = self.auth_session.get_token()
+            project_id = self.auth_session.get_project_id()
+            headers = {'X-OpenStack-Nova-API-Version': 2.11,
+                       'Content-Type': 'application/json',
+                       'X-Auth-Token': token}
```

Comment:
To add the token header you could do `headers.update(self.auth_session.get_auth_headers())` or something similar.